packages/components/dropdown/src/dropdown.vue

```diff
@@ -160,12 +161,29 @@ export default defineComponent({
     const defaultTriggerId = useId().value
     const triggerId = computed<string>(() => props.id || defaultTriggerId)
 
+    const { registerTimeout } = useTimeout()
+    const isClearTimeout = ref(false)
+    const wrapHideTimeout = computed(() => {
+      if (props.trigger === 'hover' && !isClearTimeout.value) {
+        return props.hideTimeout
+      }
+      return 0
+    })
+
     function handleClick() {
       handleClose()
```

Comment:
Maybe we can use the `hide` exposed function to close the menu directly on click.
[Ref](https://github.com/element-plus/element-plus/blob/dev/packages/components/tooltip/src/tooltip.vue#L102).

```suggestion
      popperRef.value?.hide()
```